docs/prd/command-merging.md

```
@@ -0,0 +1,246 @@
+
+    result := []Command{}
+    for each name in orderedNames {
+        result.append(commandMap[name])
+    }
+    return result
+}
+```
+
+### Precedence Chain
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)